source/common/upstream/load_stats_reporter.cc

```diff
@@ -106,27 +111,36 @@ void LoadStatsReporter::onReceiveInitialMetadata(Http::HeaderMapPtr&& metadata)
 void LoadStatsReporter::onReceiveMessage(
     std::unique_ptr<envoy::api::v2::LoadStatsResponse>&& message) {
   ENVOY_LOG(debug, "New load report epoch: {}", message->DebugString());
+  stats_.requests_.inc();
+  message_ = std::move(message);
+  prepareForResponse();
+}
+
+void LoadStatsReporter::prepareForResponse() {
   clusters_.clear();
   // Reset stats for all hosts in clusters we are tracking.
-  for (const std::string& cluster_name : message->clusters()) {
+  for (const std::string& cluster_name : message_->clusters()) {
     clusters_.emplace_back(cluster_name);
     auto cluster_info_map = cm_.clusters();
     auto it = cluster_info_map.find(cluster_name);
     if (it == cluster_info_map.end()) {
       continue;
     }
     auto& cluster = it->second.get();
+    int endpoints = 0;
     for (auto& host_set : cluster.prioritySet().hostSetsPerPriority()) {
       for (auto host : host_set->hosts()) {
         host->stats().rq_success_.latch();
         host->stats().rq_error_.latch();
+        ++endpoints;
       }
     }
     cluster.info()->loadReportStats().upstream_rq_dropped_.latch();
   }
-  stats_.requests_.inc();
+  // Disable any previous alarm.
+  response_timer_->disableTimer();
```

Comment:
Did you find this necessary for some reason? The enableTimer() call will override any previous timer value.

Reply:
Ah, cool, no I was going by [non-Envoy timer logic] here, thanks.